File: src/main/java/hu/kirdev/schpincer/web/SzorController.java
package hu.kirdev.schpincer.web;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.servlet.HandlerMapping;

import javax.servlet.http.HttpServletRequest;

@Controller
public class SzorController {

    @GetMapping({"/americano", "/pizzasch", "/reggelisch", "/kakas", "/langosch", "/vodor", "/dzsajrosz"})
    public String providerRedirects(HttpServletRequest request) {
        String providerName = (String) request.getAttribute(HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE);
        return "redirect:/p" + providerName;
    }

}

File: src/main/kotlin/hu/kirdev/schpincer/web/SzorController.kt
package hu.kirdev.schpincer.web

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.servlet.HandlerMapping
import javax.servlet.http.HttpServletRequest

@Controller
open class SzorController {

    @GetMapping(path = ["/americano", "/pizzasch", "/reggelisch", "/kakas", "/langosch", "/vodor", "/dzsajrosz"])
    fun providerRedirects(request: HttpServletRequest): String {
        val providerName = request.getAttribute(HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE) as String
        return "redirect:/p$providerName"
    }

}
